designs/NuGet-CPPCLI-Support.md

```diff
@@ -0,0 +1,115 @@
+# NuGet C++/CLI support
+
+* Status: In Review
+* Author(s): [Nikolche](https://github.com/nkolev92)
+* Issue: [8195](https://github.com/NuGet/Home/issues/8195) and [10194](https://github.com/NuGet/Home/issues/10194) NuGet support for C++/CLI projects.
+
+## Problem Background
+
+C++/CLI is a project type that allows C++ programs to use .NET classes.
+This effort tackles the NuGet side of allowing package installation of NuGet packages into these projects to take place.
+
+This effort requires coordination from NuGet, project-system and C++.
+Some of the NuGet side effort has already been done by the project-system team.
+
+Currently NuGet special cases C++ projects by looking at the extension and treating every vcxproj as packages.config `native`.
+
+The packages.config/PackageReference support story is owned by the C++ team and as of VS 16.8, all C++ projects are `packages.config`.
+
+This investigation is done in cooperation with the effort to add PackageReference support for C++/CLI scenarios, primarily targeting .Net Core.
+
+When discussing the changes here is a summary from a member on the C++ team.
+
+```md
+TargetFramework*/TargetPlatform* properties for C++/CLI are pretty much the same as for c#, so they should be used in nuget instead of just checking for .vcxproj extension. Also, C++/CLI can use/reference not only managed, but native code as well, so “native” packages should be allowed together with the appropriate managed ones.
+You can use the following properties to distinguish vc projects:
+TargetPlatformIdentifier/Version/Moniker should be defined for all vc projects.
+CLRSupport
+False or empty – native project
+NetCore – C++/CLI .NET Core project
+Everything else (true, safe, etc)  - C++/CLI .NET Framework project
+TargetFramework is only defined for C++/CLI .NET Core projects.
+TargetFrameworkIdentifier/Version/Moniker properties are defined for C++/CLI .NET Framework projects. Note, that they are also confusingly defined for pure native ones too (in msbuild common targets, at least, they used to be), so you need to check other properties first (i.e. TargetPlatformIdentifier and CLRSupport). You can also use project capabilities: “native” vs “managed”
+```
+
+## Who are the customers
+
+Customers using C++/CLI projects.
+
+## Goals
+
+Support the C++ team as necessary to provide NuGet PackageReference support for C++/CLI projects.
+
+At this point focused to:
+
+* Ensure NuGet is interpreting the correct framework.
+
+## Non-Goals
+
+* packages.config support for C++/CLI projects. Focus is on .NET Core scenarios right now.
+
+## Solution Overview
+
+In managed projects, NuGet interprets the effective framework by looking at `TargetFrameworkMoniker`, `TargetPlatformMoniker` and in UWP cases, `TargetPlatformMinVersion`.
+
+C++ projects have an additional property CLRSupport that's set for C++ projects.

```

Comment:
As per [blog](https://devblogs.microsoft.com/cppblog/porting-a-c-cli-project-to-net-core/#migrating-a-vcxproj-to-net-core), `CLRSupport` property is set to `NetCore` and `TargetFramework` set to specific `netcore tfm`. For e.g., 
```xml
<CLRSupport>NetCore</CLRSupport>
<TargetFramework>netcoreapp3.1</TargetFramework>
```
I think we need to use both `CLRSupport` and `TargetFramework` properties to enable package reference. Thoughts?

Reply:
We don't, `TargetFramework` is not a property that NuGet parses. 
For framework inference NuGet uses `TargetFrameworkMoniker` and `TargetPlatformMoniker`. 

NuGet used to use `TargetFramework` directly, but that was a problem. Now the SDK handles that.